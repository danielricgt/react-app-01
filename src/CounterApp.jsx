import PropTypes from 'prop-types';
import { useState } from "react";


const CounterApp = ({ value,
    title,
    subtitle,
    name,

}) => {
    // i destructure the props that i am passing to the component
    const [counter, setCounter] = useState(value);

    const incrementButtonEvent = () => {
        // console.log(event )
        // setCounter(counter+1); 
        setCounter(counter => (counter + 1))
        // console.log(value) 
    }

    const decrementButttonEvent = () => {
        if(counter === 0){
        return;  
    }
    else {
        setCounter(counter => (counter - 1))
    }
}

    const resetButtonEvent = () => {
        setCounter(value);
    }

    return (
        <>

            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <p>{title} {subtitle} {name}</p>
            <button onClick={incrementButtonEvent}>+1</button>
            <button onClick={decrementButttonEvent}>-1 </button>
            <button onClick={resetButtonEvent}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
}


export default CounterApp;