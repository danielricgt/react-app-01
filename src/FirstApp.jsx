// when i want to agrup two elements in a root element i can use a fragment

import { Fragment } from "react"
import PropTypes from 'prop-types';


const messageObjetc = {
    name: 'Fernando',
    age: 35,
}

const getSummarize = (a, b) => {
    return a + b;
}


export const FirstApp = ({ title='Hola, soy Dani', subtitle=123, name = 'Dani' }) => {

    if (!title) {
        throw new Error("title is required");
    }

    return (
        //* <i need always to start a component encapsulated with a father node  > */
        <>
            {/* <h1> {JSON.stringify(messageObjetc)} FirstApp!!!</h1> */}
            <h1>{title}</h1>
            <h2>{subtitle}</h2>
            <p>{name}</p> 
            {/* <p>{getSummarize(1,12)}</p> */}
        </>
    )
}

// i can use PropTypes to validate the props that i am passing to the component like a typescript strict typing


