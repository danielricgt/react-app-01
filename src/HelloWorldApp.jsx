
export const HelloWorldApp = () => (<h1>HelloWorldApp!!!</h1>)

export default HelloWorldApp;


