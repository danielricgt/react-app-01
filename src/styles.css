

html, body {
    background-color: aqua;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}
