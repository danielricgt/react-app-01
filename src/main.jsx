import React from 'react';
import ReactDOM from 'react-dom/client';

import { FirstApp } from './FirstApp';
import '../src/styles.css';
import CounterApp from './CounterApp';

// all starts with a funxpzctional component


//  now i renderize the application
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <  CounterApp value={1} name='Daniel' subtitle='Galvan' title='Go' />
    </React.StrictMode>
) 