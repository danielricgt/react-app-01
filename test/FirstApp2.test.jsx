import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('test on fist app', () => {

    const title = 'Hola, soy Daniel';
    const subtitle = 123;
    test('must match with snapshop', () => {
        const { container } = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot();
    });

    test('must show the message "Hi, im goku', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('must show the title in a  h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('must show the subtitle sent from the props', () => {
        render(<FirstApp subtitle={subtitle} />);
        expect(screen.getAllByText(subtitle).length).toBe(1 );
     });



});
