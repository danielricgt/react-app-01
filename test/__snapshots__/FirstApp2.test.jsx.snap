// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on fist app must match with snapshop 1`] = `
<div>
  <h1
    data-testid="testData"
  >
    Hola, soy Daniel
  </h1>
  <h2>
    123
  </h2>
  <p>
    Dani
  </p>
</div>
`;
